circe-akka-serializer/src/main/scala/org/virtuslab/ash/CirceAkkaSerializer.scala

```diff
@@ -0,0 +1,128 @@
+package org.virtuslab.ash
+
+import akka.actor.ExtendedActorSystem
+import akka.event.{Logging, LoggingAdapter}
+import akka.serialization.SerializerWithStringManifest
+import com.typesafe.config.Config
+import io.circe.Decoder.Result
+import io.circe.jawn.JawnParser
+import io.circe.{Decoder, Encoder, HCursor, Json, Printer}
+import org.reflections8.Reflections
+import org.virtuslab.ash.Register.Registration
+
+import java.io.NotSerializableException
+import java.lang.reflect.ParameterizedType
+import java.nio.charset.StandardCharsets.UTF_8
+import java.util.NoSuchElementException
+import scala.jdk.CollectionConverters._
+import scala.reflect.runtime.{universe => ru}
+import scala.reflect.{ClassTag, classTag}
+
+abstract class CirceAkkaSerializer[Ser <: AnyRef: ClassTag](system: ExtendedActorSystem)
+    extends SerializerWithStringManifest
+    with AkkaCodecs {
+  private val log = Logging(system, getClass)
+  private val conf = system.settings.config.getConfig("org.virtuslab.ash")
+  private val isDebugEnabled = conf.getBoolean("verbose-debug-logging") && log.isDebugEnabled
```

Comment:
Hmmm so akka debug logging must be enabled **in addition to** ASH-specific config? that might be confusing for the users (as in, _I've turned `verbose-debug-logging` on but no logs showed up_)

Reply:
(@MarconZet bump, while you're here)

Reply:
If you turn on `verbose-debug-logging` when `log.isDebugEnabled` returns false, then `log.debug()` calls don't produce any logs

Reply:
To enable debug logging you need:
1. application.conf:
```
akka.loglevel = "DEBUG"
org.virtuslab.ash {
  verbose-debug-logging = on
}
```
2. slf4j implementation configuration file:
```
org.slf4j.simpleLogger.defaultLogLevel=debug
```
Missing any of these results in no logs being produced.
Added examples to test.